updatecli/updatecli.d/jdk17.yaml

```
@@ -1,122 +1,132 @@
 ---
+# This YAML configuration file is used to bump the JDK17 version in various files and create a pull request with the changes.
+
 name: Bump JDK17 version
 
 scms:
   default:
     kind: github
     spec:
-      user: "{{ .github.user }}"
-      email: "{{ .github.email }}"
-      owner: "{{ .github.owner }}"
-      repository: "{{ .github.repository }}"
-      token: "{{ requiredEnv .github.token }}"
-      username: "{{ .github.username }}"
-      branch: "{{ .github.branch }}"
+      user: "{{ .github.user }}"  # GitHub user for authentication
+      email: "{{ .github.email }}"  # Email associated with the GitHub user
+      owner: "{{ .github.owner }}"  # Owner of the GitHub repository
+      repository: "{{ .github.repository }}"  # Name of the GitHub repository
+      token: "{{ requiredEnv .github.token }}"  # GitHub token for authentication
+      username: "{{ .github.username }}"  # GitHub username
+      branch: "{{ .github.branch }}"  # Branch to create the pull request on
 
 sources:
   lastVersion:
-    kind: githubrelease
-    name: Get the latest Adoptium JDK17 version
+    kind: temurin
+    name: Get the latest Adoptium JDK17 version via the API
```

```suggestion
    name: Get the latest Adoptium JDK17 version
```
Implementation detail, not needed.